libraries/WiFi/src/WiFiGeneric.cpp

```diff
@@ -1590,6 +1609,37 @@ int WiFiGenericClass::hostByName(const char* aHostname, IPAddress& aResult)
     return (uint32_t)aResult != 0;
 }
 
+/**
+ * Resolve the given hostname to an IP6 address.
+ * @param aHostname     Name to be resolved
+ * @param aResult       IPv6Address structure to store the returned IP address
+ * @return 1 if aHostname was successfully converted to an IP address,
+ *          else error code
+ */
+int WiFiGenericClass::hostByName6(const char* aHostname, ip_addr_t& aResult)
```

Comment:
Why is this a separate function from hostByName() ???    If I have enabled IPv6, then I just want a function to pass in a host name and get back the IPAddress (any type).  I can check the type if I want, but I should just call one function and get back the address irrespective of whether the target is IPv6 only, dual stack, or IPv4.  

The IPAddress result can handle all of these cases.

Reply:
True, this was also a cut and paste, we should revise